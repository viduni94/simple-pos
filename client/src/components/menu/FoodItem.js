import React, { Component } from "react";

class FoodItem extends Component {
  render() {
    let categoryType = this.props.category;

    return (
      <div className="col-md-6" key={categoryType._id}>
        <ul className="mu-menu-item-nav">
          <li>
            <div className="media">
              <div className="media-left">
                <img className="media-object" src={`../../img/menu/${categoryType.name}.jpg`} alt="img" />
              </div>
              <div className="media-body">
                <h4 className="media-heading">
                  <span>{categoryType.name}</span>
                </h4>
                <span className="mu-menu-price">{(categoryType.unitPrice / 100).toFixed(2)} LKR</span>
                <p>{categoryType.menu}</p>
                <p>
                  <button className="btn btn-normal btn-sm">Add Item</button>
                </p>
              </div>
            </div>
          </li>
        </ul>
      </div>
    );
  }
}

export default FoodItem;
