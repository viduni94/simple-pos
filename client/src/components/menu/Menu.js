import React, { Component } from "react";
import { getFoodItems } from "../../actions/itemActions";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import Spinner from "../common/spinner";
import FoodItem from "./FoodItem";

export class Menu extends Component {
  constructor(props) {
    super(props);

    this.addItem = this.addItem.bind(this);
    this.addItemToNewList = this.addItemToNewList.bind(this);
  }

  componentDidMount() {
    this.props.getFoodItems();
  }

  addItemToNewList = item => {
    this.props.callbackFromParent(item);
  };

  addItem = dataFromFoodItem => {
    this.addItemToNewList(dataFromFoodItem);
  };

  render() {
    const { items } = this.props.item;

    let menucontent;

    if (items === undefined) {
      menucontent = <Spinner />;
    } else {
      const beverage = items.filter(item => !item.category.localeCompare("beverages")).map(bev => <FoodItem key={bev._id} category={bev} callbackFromParent={this.addItem} />);

      const appetizer = items.filter(item => !item.category.localeCompare("appetizers")).map(app => <FoodItem key={app._id} category={app} callbackFromParent={this.addItem} />);

      const main = items.filter(item => !item.category.localeCompare("mains")).map(main => <FoodItem key={main._id} category={main} callbackFromParent={this.addItem} />);

      const dessert = items.filter(item => !item.category.localeCompare("desserts")).map(des => <FoodItem key={des._id} category={des} callbackFromParent={this.addItem} />);

      menucontent = (
        <div className="tab-content">
          <div className="tab-pane fade show active" id="beverages">
            <div className="mu-tab-content-area">
              <div className="row">{beverage}</div>
            </div>
          </div>

          <div className="tab-pane fade" id="appetizers">
            <div className="mu-tab-content-area">
              <div className="row col-md-12">{appetizer}</div>
            </div>
          </div>

          <div className="tab-pane fade" id="mains">
            <div className="mu-tab-content-area">
              <div className="row col-md-12">{main}</div>
            </div>
          </div>

          <div className="tab-pane fade" id="desserts">
            <div className="mu-tab-content-area">
              <div className="row col-md-12">{dessert}</div>
            </div>
          </div>
        </div>
      );
    }

    return (
      <section id="mu-restaurant-menu">
        <div className="container">
          <div className="row">
            <div className="col-md-12">
              <div className="mu-restaurant-menu-area">
                <div className="mu-restaurant-menu-content">
                  <ul className="nav nav-tabs mu-restaurant-menu">
                    <li>
                      <a className="nav-item nav-link active" href="#beverages" data-toggle="tab">
                        Beverages
                      </a>
                    </li>
                    <li>
                      <a className="nav-item nav-link" href="#appetizers" data-toggle="tab">
                        Appetizers
                      </a>
                    </li>
                    <li>
                      <a className="nav-item nav-link" href="#mains" data-toggle="tab">
                        Mains
                      </a>
                    </li>
                    <li>
                      <a className="nav-item nav-link" href="#desserts" data-toggle="tab">
                        Desserts
                      </a>
                    </li>
                  </ul>
                  {/* Tab panes */}
                  {menucontent}
                </div>
              </div>
            </div>
          </div>
        </div>
      </section>
    );
  }
}

Menu.propTypes = {
  item: PropTypes.object.isRequired,
  getFoodItems: PropTypes.func.isRequired
};

const mapStateToProps = state => ({
  item: state.item
});

export default connect(
  mapStateToProps,
  { getFoodItems }
)(Menu);
