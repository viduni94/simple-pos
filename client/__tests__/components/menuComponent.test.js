import React from "react";
import { shallow } from "enzyme";
import { Menu } from "../../src/components/menu/Menu";
import Spinner from "../../src/components/common/spinner";
import { wrap } from "module";

const props = {
  item: {},
  getFoodItems: jest.fn().mockReturnThis(),
  callbackFromParent: jest.fn().mockReturnThis()
};

describe("FoodItem Component", () => {
  test("renders", done => {
    const wrapper = shallow(<Menu {...props} />);
    expect(wrapper).toHaveLength(1);
    done();
  });

  test("should render initial layout", () => {
    // when
    const component = shallow(<Menu {...props} />);
    // then
    expect(component.getElements()).toMatchSnapshot();
  });

  const testProps = { ...props, item: { items: [] } };
  test("renders when there are no items", done => {
    const wrapper = shallow(<Menu {...testProps} />);
    const spinner = shallow(<Spinner />);
    expect(wrapper).toHaveLength(1);
    expect(spinner).toHaveLength(1);
    done();
  });
});
