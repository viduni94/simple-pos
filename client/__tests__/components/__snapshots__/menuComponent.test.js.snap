// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FoodItem Component should render initial layout 1`] = `
Array [
  <section
    id="mu-restaurant-menu"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-md-12"
        >
          <div
            className="mu-restaurant-menu-area"
          >
            <div
              className="mu-restaurant-menu-content"
            >
              <ul
                className="nav nav-tabs mu-restaurant-menu"
              >
                <li>
                  <a
                    className="nav-item nav-link active"
                    data-toggle="tab"
                    href="#beverages"
                  >
                    Beverages
                  </a>
                </li>
                <li>
                  <a
                    className="nav-item nav-link"
                    data-toggle="tab"
                    href="#appetizers"
                  >
                    Appetizers
                  </a>
                </li>
                <li>
                  <a
                    className="nav-item nav-link"
                    data-toggle="tab"
                    href="#mains"
                  >
                    Mains
                  </a>
                </li>
                <li>
                  <a
                    className="nav-item nav-link"
                    data-toggle="tab"
                    href="#desserts"
                  >
                    Desserts
                  </a>
                </li>
              </ul>
              <_default />
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>,
]
`;
